packages/dashboard/src/components/SecurityList.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {
+  Card,
+  CardHeader,
+  CardBody,
+  ListGroup,
+  ListGroupItem,
+  CardFooter,
+  Row,
+  Col,
+  FormSelect,
+} from 'shards-react'
+import '../../src/assets/styles.css'
+
+const gainStyles = {
+  green: '#17c671',
+  red: '#c4183c',
+}
+
+const Security = ({
+  symbol,
+  name,
+  exchange,
+  price,
+  gain,
+  currency,
+  id,
+  selectSecurity,
+  selectedSecurity,
+}) => {
+  return (
+    <>
+      <ListGroupItem
+        className={`d-flex px-3 flex-column security`}
+        style={{
+          borderTopLeftRadius: 0,
+          borderTopRightRadius: 0,
+          background: selectedSecurity?.id === id ? '#e1e5eb' : null,
+        }}
+        onClick={() => selectSecurity({ id, symbol, name, exchange, currency })}
```

Comment:
So when you select a security, it dispatches an action and gets the history for that security, then it automatically updates the mainchart with that history? how exactly does mainchart know to re-render and and how does the new chartData get in there?

Reply:
So yeah this is the handler I pass down to the onClick: 
```
  // select security from portfolio
  const selectSecurity = security => {
    setSelectedRange('1d')
    dispatch(tradeActions.selectSecurity(security))
  }
```

It sets the internal state time range and also dispatches an action, containing all that data you commented on there.

The action triggers a middleware (also called `selectSecurity`) that will ultimately call another saga (sorta confusing but this one is called `getSecurityHistory `) and that is what actually makes the request to fetch the quote data. So the data gets fetched and updates our selectedSecurity in our state (via our reducer). Because components re-render when their state or props change (and our selected security is passed from redux into our component as a prop), the component re-renders with this up to date info. 

From our component we calculate what data we need, and then pass that into the `MainChart />` component, which will plot the data into our chart. The fix that I made prevented weird re-rendering by checking the property you commented on in your first comment. That object path is the path to the actual array of data points, so we only re-render when that changes.

Chart data gets into this component via the `chartData` property which contains the chart data and options.